src/libPMacc/include/eventSystem/tasks/TaskLogicalAnd.hpp

```diff
@@ -77,34 +77,40 @@ namespace PMacc
             if (task1 == eventId)
             {
                 task1 = 0;
-                /* \todo: there is a bug in this part of code
-                 * ITask* task = Environment<>::get().Manager().getITaskIfNotFinished(task2);
+
+                ITask* task = Environment<>::get().Manager().getITaskIfNotFinished(task2);
                 if (task != NULL)
                 {
                     ITask::TaskType type = task->getTaskType();
-                    if (type == ITask::TASK_CUDA && this->getTaskType() != ITask::TASK_CUDA)
+                    if (type == ITask::TASK_CUDA )
                     {
-                        this->setTaskType(task->getTaskType());
-                        this->setCudaEvent(static_cast<StreamTask*> (task)->getCudaEvent());
+                        this->stream = static_cast<StreamTask*>(task)->getEventStream();
```

Comment:
If encapsulated (see comment in that file) one could use a protected "replaceStream" method. This would make the intention clear if this is what should happen here. (Should the stream be replaced?)